synapse/storage/databases/main/transactions.py

```diff
@@ -333,3 +333,40 @@ def _store_destination_rooms_entries_txn(
             ["stream_ordering"],
             [(stream_ordering,)] * len(rows),
         )
+
+    async def get_destination_last_successful_stream_ordering(
+        self, destination: str
+    ) -> Optional[int]:
+        """
+        Gets the stream ordering of the PDU most-recently successfully sent
+        to the specified destination.
+
+        Args:
+            destination: the destination we have successfully sent to
```

Comment:
well, we may not have successfully sent anything there...

```suggestion
            destination: the destination to query
```